### PR TITLE
Bump galaxy packages to 22.05

### PR DESCRIPTION
**Pull request by bernt-matthias**
At the moment this works .. because the 22.05 packages have not been release...

### CODE REVIEW

File: requirements.txt [hunk 1]
```diff
@@ -3,9 +3,9 @@ bioblend>=0.17.0
 click!=8.0.2
 cwltool>=1.0.20191225192155
 ephemeris>=0.10.3
-galaxy-containers<22.5
-galaxy-tool-util>=21.1.1,<22.5
-galaxy-util>=20.5.0,<22.5
+galaxy-containers<22.9
+galaxy-tool-util>=21.1.1,<22.9
+galaxy-util>=20.5.0,<22.9
```
Comment (bernt-matthias) [inline requirements.txt:8]:
```suggestion
galaxy-containers<23.0
galaxy-tool-util>=21.1.1,<23.0
galaxy-util>=20.5.0,<23.0
```

**Comment by nsoranzo** [comment]
I don't think it's time to merge this, we still don't have 22.5 packages, right?

**Comment by mvdbeek** [comment]
I pushed dev packages for util and tool_util (https://pypi.org/project/galaxy-util/22.5.0.dev0/, https://pypi.org/project/galaxy-tool-util/22.5.0.dev0/), if you want to try those.

File: planemo/galaxy/config.py [hunk 2]
```diff
@@ -26,7 +26,7 @@
     TYPE_CHECKING,
 )
 
-from galaxy.containers.docker_model import DockerVolume
+from galaxy.tool_util.deps.container_volums import DockerVolume
```
Comment (nsoranzo) [inline planemo/galaxy/config.py:29]:
```suggestion
from galaxy.tool_util.deps.container_volumes import DockerVolume
```

**Comment by nsoranzo** [comment]
> I pushed dev packages for util and tool_util (https://pypi.org/project/galaxy-util/22.5.0.dev0/, https://pypi.org/project/galaxy-tool-util/22.5.0.dev0/), if you want to try those.

FYI I have an incoming PR to update some package requirements before we release the stable 22.5 ones.

**Comment by bernt-matthias** [comment]
This one: https://github.com/galaxyproject/galaxy/pull/15159?

**Comment by nsoranzo** [comment]
> This one: [galaxyproject/galaxy#15159](https://github.com/galaxyproject/galaxy/pull/15159)?

Yes!

**Comment by bernt-matthias** [comment]
Ideally also https://github.com/galaxyproject/galaxy/pull/14572 would be included :)

**Comment by bernt-matthias** [comment]
This https://github.com/galaxyproject/galaxy/pull/15439 should fix a few of the failing tests

**Comment by mvdbeek** [comment]
I've released new dev packages with the last fixes

**Comment by bernt-matthias** [comment]
Btw. with https://github.com/galaxyproject/tools-iuc/pull/4830 we should be able to test planemo branches ... but it's a somewhat bigger PR.

File: planemo/conda.py [hunk 3]
```diff
@@ -153,7 +153,7 @@ def collect_conda_target_lists_and_tool_paths(ctx, paths, recursive=False, found
 
 def tool_source_conda_targets(tool_source):
     """Load CondaTarget object from supplied abstract tool source."""
-    requirements, _ = tool_source.parse_requirements_and_containers()
+    requirements, *_ = tool_source.parse_requirements_and_containers()
```
Comment (bernt-matthias) [inline planemo/conda.py:156]: What is this syntax doing? Never have seen this before.

File: planemo/shed_lint.py [hunk 4]
```diff
@@ -50,6 +52,21 @@
 ]
 
 
+# TODO: put this back into tool_util, my bad :(
```
Comment (bernt-matthias) [inline planemo/shed_lint.py:55]: So, this should be moved back to galaxy?

File: planemo/conda.py (hunk 3 above)
Comment (mvdbeek) [reply planemo/conda.py:156]: https://stackoverflow.com/questions/13726155/is-an-acceptable-way-to-ignore-arguments-in-python

File: planemo/shed_lint.py (hunk 4 above)
Comment (mvdbeek) [reply planemo/shed_lint.py:55]: Yep, I think we need to maintain the interface

Comment (mvdbeek) [reply planemo/shed_lint.py:55]: Yes, we can backport.

Comment (bernt-matthias) [reply planemo/shed_lint.py:55]: OK. Done https://github.com/galaxyproject/galaxy/pull/15457

File: requirements.txt [hunk 5]
```diff
@@ -3,9 +3,8 @@ bioblend>=1.0.0
 click!=8.0.2
 cwltool>=1.0.20191225192155
 ephemeris>=0.10.3
-galaxy-containers<22.5
-galaxy-tool-util>=21.1.1,<22.5
-galaxy-util>=20.5.0,<22.5
+galaxy-tool-util<23.0
```
Comment (mvdbeek) [inline requirements.txt:6]:
```suggestion
galaxy-tool-util>=21.1.1,<23.0
```

File: requirements.txt [hunk 6]
```diff
@@ -3,9 +3,8 @@ bioblend>=1.0.0
 click!=8.0.2
 cwltool>=1.0.20191225192155
 ephemeris>=0.10.3
-galaxy-containers<22.5
-galaxy-tool-util>=21.1.1,<22.5
-galaxy-util>=20.5.0,<22.5
+galaxy-tool-util>=21.1.1,<23.0
```
Comment (mvdbeek) [inline requirements.txt:6]:
```suggestion
galaxy-tool-util>=22.5.dev2,<23.0
```
I guess that's needed for the moved galaxy.tool_util.containers

File: requirements.txt [hunk 7]
```diff
@@ -3,9 +3,8 @@ bioblend>=1.0.0
 click!=8.0.2
 cwltool>=1.0.20191225192155
 ephemeris>=0.10.3
-galaxy-containers<22.5
-galaxy-tool-util>=21.1.1,<22.5
-galaxy-util>=20.5.0,<22.5
+galaxy-tool-util>=22.5.dev2,<23.0
+galaxy-util[template]>=22.1.0,<23.0
```
Comment (mvdbeek) [inline requirements.txt:7]:
```suggestion
galaxy-util[template]>=22.5.dev2,<23.0
```

**Comment by bernt-matthias** [comment]
Heureka.

**Review by nsoranzo — APPROVED** [review]

**Comment by bernt-matthias** [comment]
Any chance to get this released? Would like to get [this 2 year old PR](https://github.com/galaxyproject/tools-iuc/pull/3904) merged, but linting needs 22.05 as far as I see.

**Comment by mvdbeek** [comment]
Sure, https://pypi.org/project/planemo/0.75.5/

**Pull request merged**